lib/plaid/account.rb

```diff
@@ -1,5 +1,128 @@
+    # institutions calculate the Available Balance. In the case that Available
+    # Balance is unavailable from the institution, Plaid will either return an
+    # Available Balance value of null or only return a Current Balance.
+    attr_reader :available_balance
+
+    # Public: The Symbol institution type, e.g. :wells.
+    attr_reader :institution
+
+    # Public: The Hash with additional information pertaining to the account
+    # such as the limit, name, or last few digits of the account number. E.g.
+    # {"name": "Plaid Savings", "number": "9606" }.
+    attr_reader :meta
+
+    # Public: The String account name. Obtained from the #meta.
+    attr_reader :name
+
+    # Public: The Symbol account type. One of :depository, :credit, :loan,
+    # :mortgage, :brokerage, and :other.
+    attr_reader :type
+
+    # Public: The String account subtype. E.g. "savings".
+    #
+    # Read more about subtypes in the Plaid API docs.
+    attr_reader :subtype
+
+    # Public: The Hash with account and routing numbers for the account.
+    #
+    # The Hash contains Symbol keys and String values. E.g.
+    # {routing: "021000021", account: "9900009606", wireRouting: "021000021"}.
+    attr_reader :numbers
+
+    #attr_reader :transactions
+
+    def initialize(hash)
+      @id          = hash['_id']
+      @item_id     = hash['_item']
+      @meta        = hash['meta']
+      @name        = @meta['name'] if @meta
+      @type        = hash['type'].to_sym
+      @subtype     = hash['subtype']
+      @institution = hash['institution_type'].to_sym
+
+      if bal = hash['balance']
+        @available_balance = bal['available']
+        @current_balance   = bal['current']
+      end
+
+      @numbers = Plaid.symbolize_hash(hash['numbers'])
+    end
+
+    # Public: Get a String representation of the account.
+    #
+    # Returns a String.
+    def inspect
+      %{#<Plaid::Account id=#{id.inspect}, type=#{type.inspect}, name=#{name.inspect}, institution=#{institution.inspect}>}
+    end
+
+    # Public: Get a String representation of the account.
+    #
+    # Returns a String.
+    alias to_s inspect
+
+    # Private: Merge account information.
+    #
+    # accounts     - The Array of Account instances.
+    # new_accounts - The Array of Account instances.
+    #
+    # Returns accounts.
+    def self.merge(accounts, new_accounts)
+      # Index accounts by their ID.
+      #
+      # Same as index = accounts.index_by(&:id) in ActiveSupport.
+      index = Hash[accounts.map { |a| [a.id, a] }]
+
+      new_accounts.each do |acc|
+        if old_acc = index[acc.id]
+          old_acc.update_from(acc)
+        else
+          accounts << acc
+        end
+      end
+
+      accounts
+    end
+
+    # Private: Update account information.
+    #
+    # All fields which are not nil in another are copied to self.
+    #
+    # another - The Account instance with new information.
+    #
+    # Returns self.
+    def update_from(another)
```

Comment:
It's an internal method used by `Account.merge`. I couldn't make it a true `private` because `Account.merge` is a class function and can't access private methods.

Your point about possibly forgotten fields is correct, but I can't see better solution. 

This whole merging approach aims at accumulating information about the accounts. Example:

```ruby
user = User.create(:connect, ...)
user.auth   # adds routing information to accounts
user.balance  # updates balance, but keeps routing info in place
```
We chose this approach in a discussion with @charleswli.